packages/pyright-internal/src/languageService/signatureHelpProvider.ts

```diff
@@ -141,13 +145,32 @@ export class SignatureHelpProvider {
                 paramName = params[params.length - 1].name || '';
             }
 
+            // If we have a typedKwargs, the param name will be wrong.
+            const kwargsIndex = paramIndex >= params.length ? params.length - 1 : paramIndex;
+            const kwargsParam = params[kwargsIndex];
+            if (
+                isTypedKwargs(kwargsParam) &&
+                isClassInstance(kwargsParam.type) &&
+                kwargsParam.type.details.typedDictEntries
+            ) {
+                // Use the relative position in typed dict entries.
+                const dictIndex = paramIndex - kwargsIndex;
+                paramName = [...kwargsParam.type.details.typedDictEntries.keys()][dictIndex];
+            }
+
             parameters.push({
                 startOffset: label.length,
                 endOffset: label.length + paramString.length,
                 text: paramString,
                 documentation: extractParameterDocumentation(functionDocString || '', paramName),
             });
 
+            // Name match for active parameter as the set of parameters from the function
+            // may not match the actual string output from the typeEvaluator (kwargs for TypedDict is an example).
```

Comment:
I'm not sure what this comment means. Maybe there's a typo / wrong word in there somewhere?

Reply:
The old code would just compare the 'param' to the signature.activeParam. This doesn't work with the TypedDict stuff because there is no param that matches. 

So it has to do a match by 'name' instead.

Reply:
Any suggestion for what the comment would be to explain that is appreciated. That's what the intent of the comment was.

Reply:
Will change. Thanks for the suggestion.